### DISCUSSION TITLE
Using x-cmd with Bash in WSL2, how to make auto-suggestion and auto-completion?

### DISCUSSION THREAD

**Discussion by MaxforCherubim** [x-cmd/x-cmd#72] [Q&A] [answered]
Is it necessary to use Zsh instead of Bash, and load the zsh-plugin to make auto-suggestion and auto-completion?

**Comment by lunrenyi** [comment]
It depends on how much you rely on the auto-suggestion feature. x-cmd provides auto-completion functionality in both Bash and Zsh. 
However, in Zsh, the x-cmd theme additionally loads the zsh-syntax-highlighting and zsh-autosuggestions plugins.
- zsh-syntax-highlighting: Highlights command syntax as you type.
- zsh-autosuggestions: Provides command suggestions based on your history.
- You can disable these plugins by using the command `x theme feature use zshplugin never`.

**Reply by MaxforCherubim** [reply]
Well, when I am using Bash in Ubuntu of WSL2, seems there exists no auto-completion.
Does it need to activate this function manually?

**Reply by lunrenyi** [reply, marked as answer]
> Well, when I am using Bash in Ubuntu of WSL2, seems there exists no auto-completion. Does it need to activate this function manually?

Press the tab key after entering part of the command
For example, type `x theme up` and then press the Tab key
<img width="638" alt="image" src="https://github.com/user-attachments/assets/cf255f3e-a1a2-4029-a598-6b2cc80d9b9e">


- `export ___X_CMD_HELP_LANGUAGE=cn` sets the natural language used for completion suggestions
<img width="854" alt="image" src="https://github.com/user-attachments/assets/4698c93e-6d34-472d-a670-328c0f3d106f">

**Reply by MaxforCherubim** [reply]
Thanks for replying.
I have known what is auto-completion.

**Comment by lunrenyi** [comment]
The enhanced experience brought by zsh-autosuggestions can be partially achieved with fzf. -- [fzf 1min guide](https://www.x-cmd.com/1min/fzf#fzf)
```
command -v fzf || x env use fzf                   # ensure have fzf cmd
source <(x fzf --bash)
<ctr-r>
```
<img width="1010" alt="image" src="https://github.com/user-attachments/assets/c848c6db-f5ec-41d8-b5bd-1b139231e580">
